vortex-dtype/src/ptype.rs

```diff
@@ -48,6 +48,10 @@ pub trait NativePType:
     const PTYPE: PType;
 
     fn is_nan(self) -> bool;
+
+    fn compare(self, other: Self) -> Ordering;
+
+    fn is_eq(self, other: Self) -> bool;
```

Comment:
Can't these two be expressed with additional trait bounds?

Reply:
We can define an additional traits but afaik there isn't one that has the behaviour as implemented here

Reply:
Isn't this just `Ord`?

Reply:
ok I got it, its about the custom implementation